src/commands/data/import/bulk.ts

```diff
@@ -0,0 +1,246 @@
+      jsonEnabled: flags.json ?? false,
+      stages: async ? ['Creating ingest job'] : ['Creating ingest job', 'Processing the job'],
+      title: async ? 'Importing data (async)' : 'Importing data',
+      stageSpecificBlock: [
+        {
+          stage: 'Processing the job',
+          label: 'Processed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsProcessed) {
+              return data.numberRecordsProcessed.toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Successful records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return (data.numberRecordsProcessed - numberRecordsFailed).toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Failed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsFailed) {
+              return data.numberRecordsFailed.toString();
+            }
+          },
+        },
+      ],
+      postStagesBlock: [
+        {
+          label: 'Status',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) => data?.state,
+        },
+        {
+          label: 'Job Id',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) =>
+            data?.id &&
+            terminalLink(
+              data.id,
+              `${baseUrl}/lightning/setup/AsyncApiJobStatus/page?address=${encodeURIComponent(`/${data.id}`)}`
+            ),
+        },
+      ],
+    });
+
+    if (async) {
+      ms.goto('Creating ingest job');
+
+      const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+      ms.goto('Creating ingest job', job.getInfo());
+      ms.stop();
+
+      const cache = await BulkImportRequestCache.create();
+      await cache.createCacheEntryForRequest(job.id, conn.getUsername(), conn.getApiVersion());
+
+      this.log(messages.getMessage('export.resume', [job.id]));
+
+      return {
+        jobId: job.id,
+      };
+    }
+
+    // synchronous flow
+    const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+    ms.goto('Processing the job');
+
+    job.on('inProgress', (res: JobInfoV2) => {
+      ms.goto('Processing the job', res);
+    });
+
+    try {
+      await job.poll(5000, timeout.milliseconds);
+
+      const jobInfo = job.getInfo();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
+
+      if (jobInfo.numberRecordsFailed) {
+        ms.stop('failed');
```

Comment:
just fyi, you could also use `ms.error()` here - it does the exact the same thing as `ms.stop('failed')`

https://github.com/oclif/multi-stage-output/blob/main/src/multi-stage-output.tsx#L316